### PR TITLE
FIX: Circle timeout after plot_visualize_stc

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
I realized that since #7612 has been merged, Circle reaches timeout after `plot_visualize_stc`. I can't reproduce locally so I decided to investigate in here.

- [x] Find that the issue is related to the wrapping of `_Brain.save_movie`
- [ ] Resolve `save_movie` reference cycle (waiting for CircleCI)
- [x] Fix `UnboundLocalError`: `playback_speed_slider` used without being declared

### CODE REVIEW

**Comment by larsoner** [comment]
@GuillaumeFavelier I think it's more likely a general memory problem. Sometimes it dies on `plot_visualize_stc` but also sometimes on `plot_mne_solutions`. Here is the `mprof run plot_mne_solutions.py`:

Screenshot from 2020-04-20 08-54-39

And `plot_visualize_stc`:

Screenshot from 2020-04-20 08-58-16

It's not the same as what happens on CircleCI, because Sphinx-gallery will close and GC at the end of each plotting block. And the SG report confirms this:

https://mne.tools/dev/auto_tutorials/source-modeling/sg_execution_times.html

Same on this PR, so maybe it does indeed fix it (?):

https://19456-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/source-modeling/sg_execution_times.html

In any case, this all indicates that the memory usage of `_Brain` can be high so we need to make sure it gets closed and GC'ed properly.

**Comment by GuillaumeFavelier** [comment]
So if I understand correctly you think that the failures on CircleCI are not related to memory but we should pay attention to it? I can open a PR about that and make it my priority. Using those 2 scripts as benchmarks. What do you think?

**Comment by larsoner** [comment]
> So if I understand correctly you think that the failures on CircleCI are not related to memory but we should pay attention to it?

Either that or somehow they are related to memory and this PR fixes it (?). I have no idea why they would, though. `copy_doc` should work with `str` so it should make no reference cycles and having `_Brain` in the space doesn't seem like it should matter. Any idea why this PR seems to fix things?

Also should we merge it?

**Comment by GuillaumeFavelier** [comment]
I am not sure myself. I wanted to investigate more actually:

- restoring `copy_doc`
- restoring the original function of `_Brain` during `clean`

**Comment by GuillaumeFavelier** [comment]
Any way of getting more information really.

File: mne/viz/_brain/_timeviewer.py [hunk 1]
```diff
@@ -362,8 +361,6 @@ def __init__(self, brain, show_traces=False):
         # Direct access parameters:
         self.brain = brain
         self.brain.time_viewer = self
-        self.brain._save_movie = self.brain.save_movie
-        self.brain.save_movie = self.save_movie
```
Comment (larsoner) [inline mne/viz/_brain/_timeviewer.py:366]: Ahh, actually this could create a reference cycle that it has trouble resolving

**Comment by GuillaumeFavelier** [comment]
I will push a hotfix for the missing slider too.

**Comment by GuillaumeFavelier** [comment]
I will push a hotfix for the missing slider too.

**Comment by GuillaumeFavelier** [comment]
The patch did not work. I suggest to revert back to something similar to edb8ccb3b9cad1e48b98eef72be60cbc1efc26e4 and merge with master. 

I will find a way to put this feature back in another PR.

**Comment by GuillaumeFavelier** [comment]
This is now ready to merge @larsoner

**Comment by larsoner** [comment]
Thanks @GuillaumeFavelier !

**Pull request merged**